src/peft/tuners/lora/config.py

```diff
@@ -224,6 +228,21 @@ class LoraConfig(PeftConfig):
             )
         },
     )
+    # Enables replicating layers in a model to expand it to a larger model.
+    layer_replication: Optional[list[tuple[int, int]]] = field(
+        default=None,
+        metadata={
+            "help": (
+                "This enables using LoRA to effectively expand a transformer model to a larger size by repeating some layers. "
+                "The transformation expects a `layers` module list in the model which it modifies to expand the number of modules. "
+                "Base weights are shared so the memory usage is close to the original model."
+                "The format is a list of [start, end) pairs which specify the layer ranges to stack. For example:\n"
```

Comment:
I think it should be added here that these layers should be targeted with LoRA, otherwise it makes no sense to replicate the layers. Also, it would be great to link to some kind of resource to find out more about this. Since there is no paper, maybe the repo you posted earlier?